storage/storage.go

```diff
@@ -348,3 +395,205 @@ func EncodeName(name string) string {
+	if err != nil {
+		fmt.Println(err)
+		return false, ""
+	}
+	err = zipWriterWin64Client.Close()
+	if err != nil {
+		fmt.Println(err)
+		return false, ""
+	}
+
+	//Write to mod_link and upload new smaller smod file
+	if LinuxServer {
+		key := fmt.Sprintf("/mods/%s/%s.smod", modID, cleanName+"-LinuxServer-"+modVersion)
+		platform := "LinuxServer"
+
+		err = WriteModLink(ctx, key, versionID, platform, bufLinuxServer)
+		if err != nil {
+			fmt.Println(err)
+			return false, ""
+		}
+	}
+
+	if Win64Server {
+		key := fmt.Sprintf("/mods/%s/%s.smod", modID, cleanName+"-Win64Server-"+modVersion)
+		platform := "Win64Server"
+
+		err = WriteModLink(ctx, key, versionID, platform, bufWin64Server)
+
+		if err != nil {
+			fmt.Println(err)
+			return false, ""
+		}
+	}
+	if Win64Client {
+		key := fmt.Sprintf("/mods/%s/%s.smod", modID, cleanName+"-WindowsNoEditor-"+modVersion)
+		platform := "WindowsNoEditor"
+
+		err = WriteModLink(ctx, key, versionID, platform, bufWin64Client)
+
+		if err != nil {
+			fmt.Println(err)
+			return false, ""
+		}
+	}
+
+	key := fmt.Sprintf("/mods/%s/%s.smod", modID, cleanName+"-Combined-"+modVersion)
+	platform := "Combined"
+
+	combined := bytes.NewBuffer(body)
+
+	err = WriteModLink(ctx, key, versionID, platform, combined)
+
+	if err != nil {
+		fmt.Println(err)
+		return false, ""
+	}
+
+	key = fmt.Sprintf("/mods/%s/%s.smod", modID, cleanName+"-WindowsNoEditor-"+modVersion)
+
+	return true, key
+}
+
+func WriteZipFile(file *zip.File, platform string, zipWriter *zip.Writer) error {
+	fileName := strings.ReplaceAll(file.Name, platform+"/", "")
+	zipFile, err := zipWriter.Create(fileName)
+
+	if err != nil {
+		fmt.Println(err)
+		return fmt.Errorf("failed to create zip: %w", err)
+	}
+
+	rawFile, err := file.Open()
+
+	if err != nil {
+		fmt.Println(err)
+		return fmt.Errorf("failed to open zipped file: %w", err)
+	}
+
+	buf := new(bytes.Buffer)
+	_, err = buf.ReadFrom(rawFile)
+
+	if err != nil {
+		fmt.Println(err)
+		return fmt.Errorf("failed to read zipped file: %w", err)
+	}
+
+	_, err = zipFile.Write(buf.Bytes())
+
+	if err != nil {
+		fmt.Println(err)
+		return fmt.Errorf("failed to write file to zip: %w", err)
+	}
+
+	return nil
+}
+
+func WriteModLink(ctx context.Context, key string, versionID string, platform string, buffer *bytes.Buffer) error {
+	_, err := storage.Put(ctx, key, bytes.NewReader(buffer.Bytes()))
+
+	if err != nil {
+		fmt.Println(err)
+		return fmt.Errorf("failed to write smod: %w", err)
+	}
+
+	hash := sha256.New()
+	_, err = hash.Write(buffer.Bytes())
+
+	if err != nil {
+		fmt.Println(err)
+		return fmt.Errorf("failed to hash smod: %w", err)
+	}
+
+	dbModLink := &postgres.ModLink{
+		ModVersionLinkID: versionID,
+		Platform:         platform,
+		Key:              key,
+		Hash:             hex.EncodeToString(hash.Sum(nil)),
+		Size:             int64(len(buffer.Bytes())),
+	}
+
+	_, err = postgres.CreateModLink(ctx, dbModLink)
+
+	if err != nil {
+		fmt.Println(err)
```

Comment:
Please remove the debug println as you return an error with context

Reply:
Heads up on lint, I had to leave fmt.Errorf in due to lint... the error was...
storage/storage.go:533:10: error returned from external package is unwrapped: sig: func (*archive/zip.File).Open() (io.ReadCloser, error) (wrapcheck)
                return err
https://github.com/satisfactorymodding/smr-api/pull/12/commits/74cbf9e7fb628b477e7bb2459a0f641867ec9415